dbt/include/trino/macros/materializations/incremental.sql

```diff
@@ -1,35 +1,104 @@
-{% macro validate_get_incremental_strategy(config) %}
-  {#-- Find and validate the incremental strategy #}
-  {%- set strategy = config.get("incremental_strategy", default="append") -%}
-
-  {% set invalid_strategy_msg -%}
-    Invalid incremental strategy provided: {{ strategy }}
-    Expected one of: 'append', 'delete+insert', 'merge'
-  {%- endset %}
-  {% if strategy not in ['append', 'delete+insert', 'merge'] %}
-    {% do exceptions.raise_compiler_error(invalid_strategy_msg) %}
+{% macro get_incremental_tmp_relation_type(strategy, unique_key, language) %}
+
+  /* {#
+       If we are running multiple statements (DELETE + INSERT),
+       we must first save the model query results as a temporary table
+       in order to guarantee consistent inputs to both statements.
+
+       If we are running a single statement (MERGE or INSERT alone),
+       we can save the model query definition as a view instead,
+       for faster overall incremental processing.
+  #} */
+
+  {% if language == 'sql' and (strategy in ('default', 'append', 'merge') or (unique_key is none)) %}
+    {{ return('view') }}
+  {% else %}  {#--  play it safe -- #}
```

Comment:
remove the comment

Reply:
decided to keep it the same as in dbt-core to limit differences and make further upgrades as smooth as possible.